### PR TITLE
Removed duplicated ElasticEnergyAux auxkernel in solid mechanics and …

### LINKED ISSUE #7067: ElasticEnergyAux auxkernel migration and related
### Description of the enhancement or error report
Removed duplicated ElasticEnergyAux auxkernel in solid mechanics and updated the input file syntax in related tests 
Removed duplicated elastic_patch test in combined module

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

### PR DESCRIPTION
**Pull request by hchen139**
closes #7067
…updated the input file syntax in related tests, removed duplicated elastic_patch test in combined (#7067)

### CODE REVIEW

**Comment by dschwen** [comment]
Are we going to rename `TensorElasticEnergyAux` to `ElasticEnergyAux` sometime?

**Comment by hchen139** [comment]
@dschwen I think we can do that in this PR

File: modules/tensor_mechanics/tests/elastic_patch/tests [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
     exodiff = 'elastic_patch_out.e'
   [../]
+
+  [./test_elastic_patch_2Procs]
+    type = Exodiff
+    input = 'elastic_patch.i'
+    exodiff = 'elastic_patch_out.e'
+    min_parallel = '2'
+    prereq = 'test_elastic_patch'
+    platform = 'DARWIN'
```
Comment (dschwen) [inline modules/tensor_mechanics/tests/elastic_patch/tests]: Do we need this restriction?

Comment (hchen139) [inline modules/tensor_mechanics/tests/elastic_patch/tests]: This is from the original tests in the combined module.

Comment (dschwen) [inline modules/tensor_mechanics/tests/elastic_patch/tests]: I know. But this is not the same test anymore, so it is worth checking if the restriction is still needed.

**Comment by dschwen** [comment]
Hmm, @jasondhales this now diffs on the internal volume test. Wasn't there a change just merged to this. Do you know what is happening here?

**Comment by jasondhales** [comment]
Yes, there was a change, but I don't see how it could cause this.

The same error occurred once during one of the moosebuild tests associated with the change, which didn't make sense because there are only two rows of data and the code finished successfully.  The fact that it has happened here (twice) makes me think there is something not right with the testing, but I don't know what it is.  When I invalidated the failed moosebuild and tried again, it was successful.

If the code produced different answers (which seems very unlikely given my changes), I would suspect I introduced a bug.  But for the testing system to report that not enough data was in the csv file when the code exited properly, I don't have an explanation for that.

**Comment by permcody** [comment]
That could happen if there's a race. Are you outputting the same file from
two different tests?

**Comment by hchen139** [comment]
But the test didn't fail on local machine.

**Comment by dschwen** [comment]
Yes, it is "race related".

look at moose/modules/combined/tests/internal_volume/tests

```
  [./test_rz2]
    type = 'CSVDiff'
    input = 'internal_volume_rz.i'
    csvdiff = 'out_rz.csv'
    cli_args = 'Postprocessors/internalVolume/component=1 Outputs/file_base=out_rz'
    prereq = test_rz
  [../]

...

  [./test_quad8_rz]
    type = 'CSVDiff'
    input = 'internal_volume_rz_quad8.i'
    csvdiff = 'out_rz.csv'
    prereq = 'test_rz'
  [../]
```

Both tests write to the same output and both can run at the same time as soon as `test_rz` is done. The right thing here would be a chain dependence with `test_quad8_rz` depending on `test_rz2`.

This is one reason why I think the `file_base` parameter should absolutely be avoided in tests. And the test blocks should be named like the input files. This tests file has names all over the place.

**Comment by permcody** [comment]
Hailong, Race conditions are indeterministic which is why this test
sometimes passes and sometimes doesn't. They can be maddening to track
down. Glad this one was easy to find.


Cody

**Comment by jasondhales** [comment]
Daniel,

Good find.  This was my fault.  I didn't think that there was already a
test with a prereq.

Thanks!

-Jason

On Thu, May 26, 2016 at 8:39 AM, Daniel Schwen <notifications@github.com>
wrote:

> Merged #7068
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_7068&d=BQMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=Xc2bKK3glHdlneeCICPG9lfj75spmf8taZ0yfCJUiQA&m=cB_jJVRkS_iv1eWXhqfKZwJSmgd5qDdKZwPA8j9XEy4&s=Fu4maWGjzuLy4vqsMIXTT9ThnF6H1v_2YM7PXFzCS5Y&e=>
> .
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_7068-23event-2D673256120&d=BQMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=Xc2bKK3glHdlneeCICPG9lfj75spmf8taZ0yfCJUiQA&m=cB_jJVRkS_iv1eWXhqfKZwJSmgd5qDdKZwPA8j9XEy4&s=8KLX9Ywz40nCAwogdbBVO_v-CcNBW1hboXlmzP77Kwo&e=>
>



-- 
Jason D. Hales, PhD
Department Manager
Fuels Modeling & Simulation
Idaho National Laboratory
(208) 526-2293

File: modules/tensor_mechanics/tests/elastic_patch/tests [hunk 2]
```diff
@@ -1,7 +1,21 @@
 [Tests]
-  [./elastic_patch]
+  [./test_elastic_patch]
```
Comment (dschwen) [inline modules/tensor_mechanics/tests/elastic_patch/tests]: This is already merged, but I'm not a fan of changes like this. The user _knows_ this is a "test". There is no reason to name the test "test..".

**Pull request merged**